Tools/i18n/pygettext.py

```diff
@@ -387,7 +387,7 @@ def unparse_spec(name, spec):
             parts.append(f'{pos + 1}c')
         else:
             parts.append(str(pos + 1))
-    return f'{name}:{','.join(parts)}'
+    return f"{name}:{','.join(parts)}"
```

Comment:
f-strings support nested quotation marks, the original code is correct here
```suggestion
    return f'{name}:{','.join(parts)}'
```

Reply:
I think it is cleaner than:
```
return f'{name}:{','.join(parts)}'  # noqa: E999
```
As otherwise:
```
Tools/i18n/pygettext.py:390:22: invalid-syntax: Cannot reuse outer quote character in f-strings on Python 3.10 (syntax was added in Python 3.12)
    |
388 |         else:
389 |             parts.append(str(pos + 1))
390 |     return f'{name}:{','.join(parts)}'
    |                      ^
    |

Found 1 error.
```

Reply:
The correct solution is to set the minimum python version in the Ruff config. Source changes should be avoided where possible.